lib/chef/application.rb

```diff
@@ -108,7 +108,32 @@ def load_config_file
         config_content = config_fetcher.read_config
         apply_config(config_content, config[:config_file])
       end
+      extra_config_options = config.delete(:config_option)
       Chef::Config.merge!(config)
+      if extra_config_options
+        extra_parsed_options = extra_config_options.inject({}) do |memo, option|
+          # Sanity check value.
+          Chef::Application.fatal!("Unparsable config option #{option.inspect}") if option.empty? || !option.include?("=")
+          # Split including whitespace if someone does truly odd like
+          # --config-option "foo = bar"
+          key, value = option.split(/\s*=\s*/, 2)
+          # Call to_sym because Chef::Config expects only symbol keys. Also
+          # runs a simple parse on the string for some common types. Not using
+          # a YAML parser or similar because security. Could maybe use JSON?
+          memo[key.to_sym] = case value
```

Comment:
`YAML.parse(option).to_ruby` does work but I'm concerned this could have unexpected consequences, but then again if you can inject a value into this config option it's probably game over?